### ISSUE TITLE
Enhance "Presentation UPLOAD"  to off load cpu and start class sooner.

### ISSUE THREAD

**Issue by bigbluebutton-issue-import** [bigbluebutton/bigbluebutton#1492]
Originally reported on Google Code with ID 737

```
Enhancement to add to list for presentation uploading.

The creattion of thumbnails takes alot of cpu (GS process) and time for larger presentations
before meeting can start.

2-4 minutes for larger presentations just to get the last thumbnails.

On vmware GS pegs windows 7 to 100% for entire process..  on a modest box the cpu can
get 80-90% during process.

Suggest that the user only wait for the period to upload file to server, then immediately
show the presention once converted, with a blank already created thumbnail.. The conversion
of thumbnails could then be done in background at a lower priority, and then pushed
to clients once needed

Benefits... level the cpu when first joining conference so this spike doesnt cause
audio/video issues.

Shorter time to be begin the class in event instructor waited to last minute to upload.



```

Reported by `207TECH` on 2010-11-12 18:23:57

**Comment by ffdixon** [comment]
We're working on non-blocking upload of slides, so the presenter will be able to interact with the session without waiting for the slides to finish processing.  See: https://github.com/bigbluebutton/bigbluebutton/pull/8464

**Comment by ffdixon** [comment]
Non-blocking upload of slides is scheduled to land in BigBlueButton 2.3.

**Comment by ffdixon** [comment]
> Suggest that the user only wait for the period to upload file to server, then immediately
show the presention once converted, with a blank already created thumbnail

Implemented in 2.3.
